src/App/Fossa/Analyze.hs

```diff
@@ -330,6 +337,7 @@ uploadSuccessfulAnalysis ::
 uploadSuccessfulAnalysis (BaseDir basedir) apiOpts metadata jsonOutput override units = do
   revision <- mergeOverride override <$> (inferProjectFromVCS basedir <||> inferProjectDefault basedir)
   saveRevision revision
+  dieOnMonorepoUpload apiOpts revision
```

Comment:
[nit]: Can we move this before the `saveRevision` call?  I think it's OK either way, but seems ideal to only save if we're planning to use it.